File: sdks/go/cache_key_validation.go
package lcp

import (
	"fmt"
	"strings"

	b57 "github.com/aco/f57"
)

func validateH57CacheKey(cacheKey string, operation string) (string, error) {
	trimmed := strings.TrimSpace(cacheKey)
	if trimmed == "" {
		return "", fmt.Errorf("invalid cache_key for %s: empty", operation)
	}
	if !b57.H57IsValid(trimmed) || !b57.H57IsCanonical(trimmed) {
		return "", fmt.Errorf("invalid cache_key for %s: expected canonical H57", operation)
	}
	return trimmed, nil
}

File: sdks/nodejs/src/cache.key.validation.ts
import { h57IsCanonical, h57IsValid } from "@aco/b57-js/h57.js";

export function isH57CacheKey(cacheKey: string): boolean {
  const value = cacheKey.trim();
  if (value.length === 0) {
    return false;
  }
  return h57IsValid(value) && h57IsCanonical(value);
}

export function assertH57CacheKey(cacheKey: string, operation: string): string {
  const value = cacheKey.trim();
  if (!isH57CacheKey(value)) {
    throw new Error(`invalid cache_key for ${operation}: expected canonical H57`);
  }
  return value;
}
